src/service/resources-impl.js

```diff
@@ -362,11 +362,21 @@ export class Resources {
       this.viewport_.ensureReadyForElements();
     }
 
-    // Create and add the resource.
-    const resource = new Resource((++this.resourceIdCounter_), element, this);
+    // First check if the resource is being reparented and if it requires
+    // reconstruction. Only already built elements are eligible.
+    let resource = Resource.forElementOptional(element);
+    if (resource &&
+        resource.getState() != ResourceState.NOT_BUILT &&
```

Comment:
I'm gonna submit. There will likely be some changes to this part of going forward.